### PR TITLE
feat(#770): add count metrics automatically to empty tables

### LINKED ISSUE #770: Add "instant metrics" for projects with no metrics
### Context
Right now, when you onboard into a Lightdash project and you don't have any metrics setup, you really don't understand the value of Lightdash. 

### Goal
Users see the value of metrics on their first compile. 

Proposed solution: 
https://www.figma.com/file/AzJYFhas6F4ZCJFVZ5dpN2/instant-metrics?node-id=0%3A1

### PR DESCRIPTION
**Pull request by owlas**
* Creates metrics automatically for dbt models with no metrics defined
* Metrics are created for columns that look like "id" columns
* Metrics are `count_distinct` type
* UI shows automatic metrics with a little sparkle:
Screenshot 2021-11-11 at 15 49 47

Screenshot 2021-11-11 at 15 49 53

No option to commit auto-metrics to the project yet.

**Questions**

* @TuringLovesDeathMetal what do you think of the UI? I've realised that the "auto" bit is actually at the model level, so we could show the "instant metrics" idea at the top of the sidebar. For now I just put the visual element on each metric in the sidebar.
* @ZeRego can I get a code review? 🙏

Closes #770

### CODE REVIEW

**Comment by TuringLovesDeathMetal** [comment]
@owlas love the UI! It's simple, and the sparkles make it look different, which is great. 

I'm not sure what you mean by adding them at teh "top level"? 🤔 

One thing I'd love: once a user runs a query with a "magic metric", we have a dismissible pop up that appears (like our "Success" messages style popup) which leads them to the docs/info about metrics:
```
We noticed that you didn’t have any metrics yet, so we created some for you to play around with (_magic_ 🧚 ). 

In Lightdash, you define your metrics directly in your dbt project. 

Check out the docs and try adding some of your own! (link to docs: https://docs.lightdash.com/get-started/setup-lightdash/add-metrics)
```

**Comment by TuringLovesDeathMetal** [comment]
@owlas What about if we had a sub-heading under metrics which said `magic metrics`, then below the magic metrics, we had a subheading which said `create your own metrics` which is a hyperlink that leads you to the docs: https://docs.lightdash.com/get-started/setup-lightdash/add-metrics

I'm imagining something in light grey, italic sort of text if that makes sense?

**Review by ZeRego — APPROVED** [review]

File: packages/frontend/src/components/TableTree.tsx [hunk 1]
```diff
@@ -236,12 +236,19 @@ const TableTree: FC<TableTreeProps> = ({
...
-                            secondaryLabel: metric.source && (
-                                <NodeItemButtons
-                                    node={metric}
-                                    onOpenSourceDialog={onOpenSourceDialog}
-                                />
-                            ),
+                            secondaryLabel:
+                                (metric.isAutoGenerated && (
+                                    <Icon
+                                        style={{ width: '25px' }}
```
Comment (ZeRego) [inline packages/frontend/src/components/TableTree.tsx:242]:
```suggestion
                                        style={{ marginRight: 7 }}
```
This makes it more aligned

**Comment by owlas** [comment]
Screenshot 2021-11-12 at 14 58 12

Thanks all for feedback! Here's the final design ^

**Pull request merged**